etl/steps/data/garden/health/2023-04-18/wgm_mental_health.py

```diff
@@ -0,0 +1,339 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import List
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+from .shared import (
+    MAPPING_AGE_VALUES,
+    MAPPING_COLUMN_NAMES,
+    MAPPING_GENDER_VALUES,
+    MAPPING_QUESTION_VALUES,
+)
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+# Minimum number of participants answering a question. Questions for demographic groups with fewer answers are filtered out.
+# This mostly affects very granular breakdowns (e.g. by age and gender)
+THRESHOLD_ANSWERS = 100
+
+
+def load_countries_regions() -> Table:
+    """Load countries-regions table from reference dataset (e.g. to map from iso codes to country names)."""
+    ds_reference: Dataset = paths.load_dependency("reference")
+    tb_countries_regions = ds_reference["countries_regions"]
+
+    return tb_countries_regions
+
+
+def load_population() -> Table:
+    """Load population table from population OMM dataset."""
+    ds_indicators: Dataset = paths.load_dependency(channel="garden", namespace="demography", short_name="population")
+    tb_population = ds_indicators["population"]
+
+    return tb_population
+
+
+def run(dest_dir: str) -> None:
+    log.info("wgm_mental_health: start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("wgm_mental_health")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["wgm_mental_health"]
+
+    # Create a dataframe with data from the table.
+    df = pd.DataFrame(tb_meadow)
+
+    #
+    # Process data.
+    #
+    # Create a new table with the processed data.
+    tb_garden = make_table(df)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_meadow.metadata)
+    ds_garden.update_metadata(paths.metadata_path)
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+    log.info("wgm_mental_health: end")
+
+
+def make_table(df: pd.DataFrame) -> Table:
+    """Create a new table from a dataframe."""
+    df = clean_df(df)
```

Comment:
I'd prefer to explicitly show the main steps inside "run", instead of hiding them inside a function called `make_table`, whose name doesn't really tell much.

Reply:
Ideally, what happens inside the `run` function should be readable by anyone (even with little Python knowledge), and they should be able to understand the main process and especially the main modifications done to the data. There's an optimal point between showing too much and showing too little. In my opinion, `make_table` is rather on the side "showing too little".